zerver/management/commands/logout_all_users.py

```diff
@@ -23,13 +27,31 @@ def add_arguments(self, parser: ArgumentParser) -> None:
             action="store_true",
             help="Only log out all users who are deactivated",
         )
+        parser.add_argument(
+            "--rotate-api-keys",
+            action="store_true",
+            help="Also rotate API keys of the affected users",
+        )
         self.add_realm_args(parser, help="Only log out all users in a particular realm")
 
     def handle(self, *args: Any, **options: Any) -> None:
         realm = self.get_realm(options)
+        rotate_api_keys = options["rotate_api_keys"]
         if realm:
             delete_realm_user_sessions(realm)
+            if rotate_api_keys:
+                bulk_regenerate_api_keys(
+                    UserProfile.objects.filter(realm=realm).values_list("id", flat=True)
+                )
         elif options["deactivated_only"]:
             delete_all_deactivated_user_sessions()
+            if rotate_api_keys:
+                bulk_regenerate_api_keys(
+                    UserProfile.objects.filter(
+                        Q(is_active=False) | Q(realm__deactivated=True)
+                    ).values_list("id", flat=True)
+                )
         else:
             delete_all_user_sessions()
+            if rotate_api_keys:
+                bulk_regenerate_api_keys(UserProfile.objects.values_list("id", flat=True))
```

Comment:
This is fetching **all** user ids on the servers, do we want to batch the operation? E.g. this will all queue up a lot of events in `do_regenerate_api_key` call:
```
    event = {"type": "clear_push_device_tokens", "user_profile_id": user_profile.id}
    queue_json_publish("deferred_work", event)
```

Reply:
I think it's probably fine; this operation is not intended to be something done often, so it's OK to have it make a large backlog.